packages/reactor-api/test/benchmarks/sync.bench.ts

```diff
@@ -0,0 +1,151 @@
+  ListenerFilter,
+} from "document-model-libs/document-drive";
+import {
+  Document,
+  DocumentModel,
+  OperationScope,
+} from "document-model/document";
+
+import { beforeAll, bench, describe } from "vitest";
+
+class TestReceiver<
+  T extends Document = Document,
+  S extends OperationScope = OperationScope,
+> implements IReceiver<T, S>
+{
+  async onStrands(strands: InternalTransmitterUpdate<T, S>[]) {
+    return Promise.resolve();
+  }
+
+  async onDisconnect() {
+    return Promise.resolve();
+  }
+}
+
+beforeAll(async () => {});
+
+describe("Document Drive", async () => {
+  const documentModels = Object.values([
+    DocumentDrive,
+    RealWorldAssets,
+  ]) as DocumentModel[];
+  const document = await RealWorldAssets.utils.loadFromFile(
+    "test/data/BlocktowerAndromeda.zip",
+  );
+
+  bench(
+    "Load PHDM into Document Drive",
+    async () => {
+      const serverA = new DocumentDriveServer(
+        documentModels,
+        new MemoryStorage(),
+      );
+      await serverA.initialize();
+
+      const serverB = new DocumentDriveServer(
+        documentModels,
+        new MemoryStorage(),
+      );
+      await serverB.initialize();
+
+      const driveAId = generateUUID();
+      const documentId = generateUUID();
+
+      const driveA = await serverA.addDrive({
+        global: {
+          id: driveAId,
+          name: "Test Drive",
+          icon: null,
+          slug: null,
+        },
+        local: {
+          availableOffline: false,
+          sharingType: "PRIVATE",
+          listeners: [],
+          triggers: [],
+        },
+      });
+
+      const driveBId = generateUUID();
+      const driveB = await serverB.addDrive({
+        global: {
+          id: driveBId,
+          name: "Test Drive",
+          icon: null,
+          slug: null,
+        },
+        local: {
+          availableOffline: false,
+          sharingType: "PRIVATE",
+          listeners: [],
+          triggers: [],
+        },
+      });
+
+      // listener!
+      const filter: ListenerFilter = {
+        branch: ["*"],
+        documentId: ["*"],
+        documentType: ["*"],
+        scope: ["*"],
+      };
+
+      const receiver = new TestReceiver();
+      await serverA.addInternalListener(driveAId, receiver, {
+        listenerId: generateUUID(),
+        label: "Test Listener",
+        block: false,
+        filter,
+      });
+
+      await serverB.addInternalListener(driveBId, receiver, {
+        listenerId: generateUUID(),
+        label: "Test Listener",
+        block: false,
+        filter,
+      });
+
+      // loads document in drive A
+      const addFileAction = generateAddNodeAction(
+        driveA.state.global,
+        {
+          documentType: document.documentType,
+          id: documentId,
+          name: "BlocktowerAndromeda",
+        },
+        ["global"],
+      );
+      await serverA.addDriveAction(driveAId, addFileAction);
+
+      const result = await serverA.addOperations(
```

Comment:
The drives are not syncing here yet.

This bench seems to get stuck, or it just takes way too long.
It seems to be related to having the internal the listener. It receives the addFile action from the drive but not the document specific operations.

Reply:
Sorry, didn't include that in my notes. The `sync.bench.ts` is not yet operational.